src/modules/fancyzones/lib/Zone.cpp

```diff
@@ -106,14 +106,22 @@ void Zone::SizeWindowToZone(HWND window, HWND zoneWindow) noexcept
     }
 
     WINDOWPLACEMENT placement{};
-    ::GetWindowPlacement(window, &placement);
-    placement.rcNormalPosition = zoneRect;
-    placement.flags |= WPF_ASYNCWINDOWPLACEMENT;
+    
     // Do not restore minimized windows. We change their placement though so they restore to the correct zone.
     if ((placement.showCmd & SW_SHOWMINIMIZED) == 0)
     {
+        //wait if SW_SHOWMINIMIZED would be removed from window (Issue #1685)
+        for (int i = 0; i < 5; i++)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+            ::GetWindowPlacement(window, &placement);
```

Comment:
In the loop we should check if `SW_SHOWMINIMIZED` has been changed in in that case stop looping.